Python/perf_trampoline.c

```diff
@@ -491,10 +449,6 @@ _PyPerfTrampoline_Fini(void)
         tstate->interp->eval_frame = NULL;
     }
     free_code_arenas();
-    if (trampoline_api.state != NULL) {
-        trampoline_api.free_state(trampoline_api.state);
-        trampoline_api.state = NULL;
-    }
```

Comment:
How about using the same thing as the init?
```c
    if (trampoline_api.state != NULL && trampoline_api.free_state!= NULL) {
        trampoline_api.free_state(trampoline_api.state);
        trampoline_api.state = NULL;
    }
```

Reply:
Keeping in mind that this particular backend for the trampoline API explicitly sets free_state to NULL, I don't quite see a benefit to spending cycles checking it. Can you elaborate on why this is better?

In any case, this will be a no-op, so I don't mind doing this change if there's a benefit to it :)

Reply:
If the `init_state` is optionally called, why should we not also call the `free_state`? I don't see why we should have this asymmetry.

If both `init_state` and `fini_state` are no-op, maybe we can remove both from `trampoline_api`? 

----
`fini_state` is now an unused variable. I think either we should use it, or we should remove it.

Reply:
That's an interesting suggestion. Just to make sure I understand correctly - this involves changing the `trampoline_api_st` struct, which is a member of the `_ceval_runtime_state` struct?

Going by the [discussion in the initial implementation](https://github.com/python/cpython/pull/96123#pullrequestreview-1079722071), this API was designed to support different profiling backends, with the `perf` tool being one of them. In the future, I expect we might use this struct to support profiling tools that do require some state initialization and cleanup, for which this is useful to have. 

Changing the `trampoline_api_st` probably warrants a separate discussion. The scope of this particular PR is limited to:
1. Introducing C-APIs for managing the perf-map file.
2. Changing the `perf_trampoline.c` implementation to use the above APIs

Maybe @pablogsal or @gpshead might have more comments on this :)

Reply:
Forgot to mention: I share your concern about the asymmetry here, thanks for noticing. We could get rid of the `init_state` call here, since that function is always NULL for the `perf` backend (y)